native/c/zds.cpp

```diff
@@ -748,6 +750,459 @@ void load_recfm_from_dscb(const DSCBFormat1 *dscb, string *recfm)
+  uint8_t dspac = (scal1 & 0xC0) >> 6;
+  bool use_cylinders = (dspac == 0x03);
+
+  // Check if this is a PDSE
+  bool is_pdse = (dscb->ds1smsfg & DS1PDSE_MASK) != 0;
+
+  if ((dscb->ds1dsorg & DS1DSGDA_MASK) || (dscb->ds1dsorg & DS1ACBM_MASK) || is_pdse)
+  {
+    // DA (Direct Access/BDAM), VSAM, or PDSE: DS1LSTAR not meaningful
+    entry.usedp = -1;
+    entry.usedx = -1;
+    return;
+  }
+
+  // Parse DS1LSTAR (last used track in TTR format)
+  uint8_t lstar_tt_high = static_cast<unsigned char>(dscb->ds1lstar[0]);
+  uint8_t lstar_tt_low = static_cast<unsigned char>(dscb->ds1lstar[1]);
+  uint32_t last_used_track = (lstar_tt_high << 8) | lstar_tt_low;
+
+  // Check if this is a large format dataset that uses DS1TTHI
+  bool is_large = (dscb->ds1flag1 & 0x10) != 0;
+  if (is_large)
+  {
+    uint8_t lstar_tt_highest = static_cast<unsigned char>(dscb->ds1ttthi);
+    last_used_track |= (lstar_tt_highest << 16);
+  }
+
+  // Special case: DS1LSTAR = 0x000000 for blksize=0 means unused
+  bool is_blksize_zero = (entry.blksize == 0);
+  if (last_used_track == 0 && is_blksize_zero)
+  {
+    entry.usedp = 0;
+    entry.usedx = 0;
+    return;
+  }
+
+  // Store used space temporarily in tracks/cylinders (will convert to percentage later)
+  if (use_cylinders)
+  {
+    entry.usedp = (last_used_track + 1 + 14) / 15; // Round up to cylinders
+  }
+  else
+  {
+    entry.usedp = last_used_track + 1; // Tracks (0-based, so +1)
+  }
+
+  // Calculate how many extents contain data (count of used extents)
+  if (entry.usedp > 0)
+  {
+    const char *extent_data = dscb->ds1exnts;
+    int cumulative_tracks = 0;
+    entry.usedx = 0;
+
+    for (int i = 0; i < 3; i++)
+    {
+      const char *extent = extent_data + (i * 10);
+      uint8_t extent_type = static_cast<uint8_t>(extent[0]);
+
+      if (extent_type == 0x00)
+        break;
+
+      int tracks_in_extent = calculate_extent_tracks(extent);
+      cumulative_tracks += tracks_in_extent;
+
+      // Count this extent as used if it contains data up to the last used track
+      if (last_used_track < cumulative_tracks)
+      {
+        entry.usedx = i + 1;
+        break;
+      }
+    }
+  }
+
+  // Calculate used percentage
+  // Note: DS1TRBAL contains bytes remaining calculated by IBM TRKCALC,
+  // but since we don't have access to TRKCALC's exact capacity formula,
+  // we approximate by using simple track-based percentage calculation.
+
+  if (entry.spacu == "BYTES" && entry.blksize > 0)
+  {
+    if (entry.usedp > 0 && entry.alloc > 0)
+    {
+      int blocks_per_track = 56664 / entry.blksize;
+      if (blocks_per_track <= 0)
+        blocks_per_track = 1;
+      int base_bytes_per_track = blocks_per_track * entry.blksize;
+
+      int used_tracks = entry.usedp;
+      int total_tracks = entry.alloc / base_bytes_per_track;
+
+      if (total_tracks > 0)
+      {
+        double percentage = (100.0 * used_tracks) / total_tracks;
+        entry.usedp = (int)percentage;
```

Comment:
I recall discussing offline that the truncation is done by z/OSMF for percentage calculation, which is why we're truncating here. Thoughts on adding a small comment explaining why we always truncate the value? To a dev without context, this seems like a possible truncation error.

Reply:
Added comment in https://github.com/zowe/zowe-native-proto/pull/625/commits/fd46a4ce19a93898697d926f316dee6ffba13b0b